metatensor-torch/tests/utils/mod.rs

```diff
@@ -54,51 +58,110 @@ fn find_python() -> PathBuf {
     panic!("could not find Python 3")
 }
 
-/// Create a fresh Python virtualenv, to install dependencies we get from Python
-/// for the tests
-pub fn create_python_venv(build_dir: PathBuf) -> PathBuf {
-    // create virtual env
-    let status = Command::new(find_python())
-        .arg("-m")
-        .arg("venv")
-        .arg(&build_dir)
-        .status()
-        .expect("failed to run python");
-    assert!(status.success(), "failed to run `python -m venv`");
-
-    let mut python = build_dir;
+/// Helper: get python executable path inside a venv
+fn python_in_venv(venv_dir: &Path) -> PathBuf {
+    let mut python = venv_dir.to_path_buf();
     if cfg!(target_os = "windows") {
         python.extend(["Scripts", "python.exe"]);
     } else {
         python.extend(["bin", "python"]);
     }
+    python
+}
 
-    // update pip in case the system uses a very old one
-    let status = Command::new(&python)
-        .arg("-m")
-        .arg("pip")
-        .arg("install")
-        .arg("--upgrade")
-        .arg("pip")
-        .status()
-        .expect("failed to run python");
-    assert!(status.success(), "failed to run `python -m pip install --upgrade pip`");
+/// Create a fresh Python virtualenv using uv if available, else fallback to python -m venv
+pub fn create_python_venv(build_dir: PathBuf) -> PathBuf {
+    if let Some(uv_bin) = find_uv() {
+        let status = Command::new(&uv_bin)
+            .arg("venv")
+            .arg(&build_dir)
+            .status()
+            .expect("failed to run uv venv");
+        assert!(status.success(), "failed to create venv using uv");
```

Comment:
this warns with `warning: A virtual environment already exists at `/home/runner/work/metatensor/metatensor/target/tmp/torch-tests/deps/torch`. In the future, uv will require `--clear` to replace it` on CI, should we pass the flag?